### PR TITLE
Support for Python 3.11

### LINKED ISSUE #1446: Support Python 3.11
Requires a new Kedro release that includes https://github.com/kedro-org/kedro/pull/2742 (already merged)

See also https://github.com/kedro-org/kedro-plugins/issues/265, https://github.com/kedro-org/kedro/issues/2270

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1446 

## Development notes

1. Included python 3.11 in the python_version matrix for build_backend job in CircleCI
2. Modified make file commands (`run` and `pytest`) to include [PYTHONWARNINGS](https://github.com/kedro-org/kedro/pull/2742#issuecomment-1619966480)
3. Bumped python version to include 3.11 in the kedro-viz package setup

## QA notes

1. Navigate to kedro-viz project folder 
2. Checkout branch - `feature/support-python_3.11`

**Local conda env setup for python 3.11 -**
  - Create a conda environment with python 3.11 - 
  
    ```shell
    conda create -n yourenvname python=3.11
    ```  
  - Activate the created conda environment - 
  
     ```shell
    conda activate yourenvname
    ```  

**Testing kedro-viz dev server -**

1. Install local kedro-datasets package by checking out kedro-plugins branch (`bump/python-3.11-support`) and executing the command -
 
    ```python
    pip install -e kedro-datasets
    ``` 

**Note**: Since the kedro-datasets are not yet released to support py3.11, we are installing local version

2. Install package dependencies for kedro-viz -

      ```python
    pip3 install -r package/test_requirements.txt -r demo-project/src/docker_requirements.txt
    ``` 

3. Run backend server by executing the command - 
     
     ```shell
    make run
    ```

4. Run frontend by executing the command -

    ```shell
    npm start
    ```

**Testing kedro-viz as a package -**

1. Build frontend

   ```shell
    make build    
    ```
    
2. Install local Kedro-viz package - 

    ```python
    pip install -e package
    ```

3. Navigate to demo-project -  
  
    ```shell
    cd demo-project
    ```
4. Execute kedro viz command to open up in the browser [here](http://127.0.0.1:4141/)
 
   ```shell
    kedro viz
    ```

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1502"></a>

### CODE REVIEW

File: package/setup.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
     long_description_content_type="text/markdown",
     license="Apache Software License (Apache 2.0)",
     url="https://github.com/kedro-org/kedro-viz",
-    python_requires=">=3.7, <3.11",
+    python_requires=">=3.7, <3.12",
```
Comment (astrojuanlu) [inline package/setup.py:39]: For the record, Kedro and kedro-datasets are now lifting the upper bound completely.

If Kedro-Viz has a strong feeling against protecting users from installing it on a version that is too new, you might want to add a warning like we did in https://github.com/kedro-org/kedro/pull/2742

Bear in mind that Python 3.12 is out two months from now

**Comment by noklam** [comment]
I saw `kedro-datasets` installation is failing. It may require install from source temporarily for now because we have a deadlock if `kedro` don't release first  `kedro-dataset` won't be available.

Do we need to discuss this offline?

**Comment by astrojuanlu** [comment]
If I understand correctly, after merging https://github.com/kedro-org/kedro-plugins/pull/297 and releasing kedro-datasets, everything should be fine. kedro can be already installed on Python 3.11. Am I missing something?

**Comment by ravi-kumar-pilla** [comment]
> I saw `kedro-datasets` installation is failing. It may require install from source temporarily for now because we have a deadlock if `kedro` don't release first `kedro-dataset` won't be available.
> 
> Do we need to discuss this offline?

Hi Nok, Happy to discuss, but Kedro-Viz can wait for Kedro-datasets to be released. @rashidakanchwala any thoughts on this ?

**Review by rashidakanchwala — APPROVED** [review]
LGTM

**Review by NeroOkwa — APPROVED** [review]

**Pull request merged**